snap/hooktypes.go

```diff
@@ -35,6 +35,7 @@ var supportedHooks = []*HookType{
 	NewHookType(regexp.MustCompile("^connect-(?:plug|slot)-[-a-z0-9]+$")),
 	NewHookType(regexp.MustCompile("^disconnect-(?:plug|slot)-[-a-z0-9]+$")),
 	NewHookType(regexp.MustCompile("^check-health$")),
+	NewHookType(regexp.MustCompile("^fde-setup$")),
```

Comment:
I haven't seen the main fde branch yet, but perhaps this should have a comment explaining why is it special?

Reply:
It's not really special, I will propose more and more PRs that build this out and use it. Having this early and isolated makes things easier for testing manually for the Taipei team because they don't have to bind mount snap-exec inside the core/snapd snap anymore once this is merged because then snap-exec considers fde-setup a valid hook.